dot/rpc/modules/state.go

```diff
@@ -47,13 +46,44 @@ type StateStorageKeyRequest struct {
 }
 
 // StateRuntimeMetadataQuery is a hash value
-type StateRuntimeMetadataQuery common.Hash
+type StateRuntimeMetadataQuery struct {
+	Bhash common.Hash
```

Comment:
is this able to accept `nil` or only `""` ? I think it needs to be able to accept nil

Reply:
same for other request types

Reply:
I have updated the PR. Now `common.Hash` accepts nil and `""` value both.